src/app/lib/analyticsUtils/index.js

```diff
@@ -149,3 +149,45 @@ export const getPublishedDatetime = (attribute, data) => {
     ? getISODate(publishedDatetime)
     : null;
 };
+
+export const getAtiUrl = data => {
+  const cleanedValues = data.filter(({ value }) => value);
+
+  const parsedAtiValues = cleanedValues.map(({ key, value, wrap }) =>
+    wrap ? `${key}=[${value}]` : `${key}=${value}`,
+  );
+
+  return parsedAtiValues.join('&');
+};
+
+export const getPublishingInfo = (service, eventInfo, eventType) => {
+  if (eventType === 'click') {
+    /*
+    https://paper.dropbox.com/doc/Event-tracking--AjJpWibjeQPWsoRLFdZW13Y9Ag-3i47TvVb9IJBMJFcL8D6u
+    for click events we need to know:
+      * campaings = [service-name]-[component], e.g. yoruba-navigation
+      * creation (label) = [component]-[item], e.g. navigation-ere_idaraya
+      * creation (type) = [event-type], e.g. click or background
+      * url = the destination link
+      * format = [PAR=container-[component-name]::name~CHD=slot] = [PAR=container-navigation::name~CHD=1]
+         ^ for format, currently using the name of the component clicked as slot
+    */
+    const eventComponentInfo = eventInfo.path[0].dataset.info || 'brand-top'; // psammead components need to be updated with 'data' attrs
+    const cleanCompInfo = eventComponentInfo.split('/').join('-');
+    const componentName = 'navigation'; // psammead components need to be updated with 'data' attrs
+    const format = `PAR=container-${componentName}::name~CHD=${cleanCompInfo.toLowerCase()}`;
+    const url = eventInfo.srcElement.href || '/';
+
+    return `PUB-[${service}-${componentName}]-[${eventInfo.type}]-[]-[${format}]-[]-[]-[]-[${url}]`;
+  }
+
+  // for background events we just need "campaings" => [service-name]-[component-name], e.g. yoruba-navigation
+  const componentName = eventType[0].dataset.info || 'navigation'; // psammead components need to be updated with 'data' attrs
+
+  return `PUB-[${service}-${componentName}]`;
+};
+
+export const getComponentsToTrack = trackClassName => {
+  // should probably find a better way of doing this
+  return document.getElementsByClassName(trackClassName);
```

Comment:
Can we do this via data attributes please? Will have a think for a better alternative in the meantime